sdk/core/azure-core/inc/azure/core/http/policies/policy.hpp

```diff
@@ -581,6 +582,9 @@ namespace Azure { namespace Core { namespace Http { namespace Policies {
       BearerTokenAuthenticationPolicy(BearerTokenAuthenticationPolicy const& other)
           : BearerTokenAuthenticationPolicy(other.m_credential, other.m_tokenRequestContext)
       {
+        std::shared_lock<std::shared_timed_mutex> readLock(other.m_accessTokenMutex);
```

Comment:
Nice catch.

Reply:
What's the catch here? If this reveals a concurrency bug, can we add a unit test which catches it?

Reply:
We could, but that is extra cost, we'll need test hooks for that, similar how token cache has them:
https://github.com/Azure/azure-sdk-for-cpp/blob/217b3cd8a53c45347a4e2fee052da29a5c042021/sdk/identity/azure-identity/inc/azure/identity/detail/token_cache.hpp#L28-L41

Feel free to create a work item, they can be implemented.

That could be an uber item for several of such places that use mutexes - i.e. we could use the same approach for the mutex in `ChainedTokenCredential` that you added last month.